lib/modules.nix

```diff
@@ -1036,6 +1036,17 @@ let
   defaultOrderPriority = 1000;
   mkAfter = mkOrder 1500;
 
+  /*
+    For use with options of type `attrsOf *`, `lazyAttrsOf *`, or `listOf *`.
+    Accepts a value that is merged with every element defined elsewhere. May
+    also accept a function that takes the attribute name or index (1-based),
```

Comment:
This thread preallocated for bikeshedding the choice of 1-based indexing.

I made an arbitrary choice here but AFAIK (not far) Nixpkgs is somewhat incoherent in which of these it uses by default. `imap` defaults to 1-indexed (but has explicit `imap0` and `imap1` options), `elemAt` defaults to 0-indexed (and has no `elemAt1` alternative). Values displayed to users seem to be 1-indexed.